README.md

```diff
@@ -0,0 +1,303 @@
+# Silverstripe CMS RevoltEvent Dispatcher Module (experimental)
+[![CI](https://github.com/archiprocode/silverstripe-revolt-event-dispatcher/actions/workflows/ci.yml/badge.svg)](https://github.com/archiprocode/silverstripe-revolt-event-dispatcher/actions/workflows/ci.yml)
+
+This module adds the ability to dispatch and listen for events in Silverstripe CMS. It's built around Revolt PHP
+and AMPHP. It aims to process events asynchronously. It also provides some abstraction to help managing event around
+common DataObject operations.
+
+## Installation
+
+```bash
+composer require archipro/silverstripe-revolt-event-dispatcher
+```
+
+## Features
+- Automatic event dispatching for DataObject operations (create, update, delete)
+- Support for versioned operations (publish, unpublish, archive, restore)
+- Asynchronous event handling using Revolt Event Loop
+
+## Setting up the Event Loop
+
+Because we are using Revolt PHP, you need to run the event loop to process the events.
```

Comment:
I wouldn't be surprised if we come up with a use case where we actually need to just wait for the event, instead of it being completely deferred. Unless your intention for this library is to be always deferred, with no way to one day decide "I need to use this event.. now". 

There's also being able chain reactions to events.. like:
```php
Event::queueEvent()->map(function ($result) {
// You can now do something else with the event.. and it's still deferred
});
```

There'a also

```php
Event::queueEvent()->catch(function ($exception) {
// Maybe I expect this event to fail at times and want to run something else in case
});
```

and then of course, finally... for use cases that require an event to be acted on immediately:
```php
$result = Event::queueEvent()->await();
```
for whatever reason. I guess the argument is to just use whatever internal service is being called directly, but, I'd be too lazy to do that and just want to deal with it then and there with the code i have.